src/snowflake/snowpark/session.py

```diff
@@ -569,6 +579,10 @@ def cte_optimization_enabled(self) -> bool:
         """
         return self._cte_optimization_enabled
 
+    @property
```

Comment:
we should have a set function?

Reply:
yes, added, aslo added a telemetry on variable setting